### PR TITLE
refactor: [RC] start migration automatically if it was done in and older version and add manual trigger

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -831,4 +831,6 @@
     <string name="cancel_login_dialog_title">Are you sure you want to cancel?</string>
     <!--    session reset-->
     <string name="label_system_message_session_reset">%1$s was unable to decrypt some of your messages but has solved the issue. This affected all conversations you share together.</string>
+    <string name="label_manual_migration_title">Migrating From Old App</string>
```
Comment (tmspzz) [inline app/src/main/res/values/strings.xml]: Can you change this to migrate from version `3.82.38`

**Review by tmspzz — APPROVED** [review]

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -831,4 +831,6 @@
...
+    <string name="start_manual_migration">Start migration form old App maually</string>
```
Comment (tmspzz) [inline app/src/main/res/values/strings.xml]: same here, Old App = version `3.82.38`

File: app/src/main/kotlin/com/wire/android/datastore/GlobalDataStore.kt [hunk 3]
```diff
@@ -89,6 +89,6 @@ class GlobalDataStore @Inject constructor(@ApplicationContext private val contex
      * If there is no status stored, the status will be [UserMigrationStatus.NoNeed]
      * meaning that the user does not need to be migrated.
      */
-    fun getUserMigrationStatus(userId: String): Flow<UserMigrationStatus> =
-        context.dataStore.data.map { UserMigrationStatus.fromInt(it[userMigrationStatusKey(userId)] ?: UserMigrationStatus.NoNeed.value) }
+    fun getUserMigrationStatus(userId: String): Flow<UserMigrationStatus?> =
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/datastore/GlobalDataStore.kt:92]: this change will make the app return null on `ShouldTriggerMigrationForUserUserCase` which will trigger migration again

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1503/4/) **failed**.

**Review by tmspzz — APPROVED** [review]

**Pull request merged**